sympy/physics/mechanics/joint.py

```diff
@@ -1378,3 +1380,431 @@ def _set_linear_velocity(self):
+
+    """
+
+    def __init__(self, name, parent, child, rotation_coordinate=None,
+                 planar_coordinates=None, rotation_speed=None,
+                 planar_speeds=None, parent_point=None, child_point=None,
+                 parent_interframe=None, child_interframe=None,
+                 planar_vectors=None, rotation_axis=None):
+        self._rotation_axis = rotation_axis
+        self._planar_vectors = planar_vectors
+        coordinates = (rotation_coordinate, planar_coordinates)
+        speeds = (rotation_speed, planar_speeds)
+        super().__init__(name, parent, child, coordinates, speeds,
+                         parent_point, child_point,
+                         parent_interframe=parent_interframe,
+                         child_interframe=child_interframe)
+
+    def __str__(self):
+        return (f'PlanarJoint: {self.name}  parent: {self.parent}  '
+                f'child: {self.child}')
+
+    @property
+    def rotation_coordinate(self):
+        """Generalized coordinate corresponding to the rotation angle."""
+        return self.coordinates[0]
+
+    @property
+    def planar_coordinates(self):
+        """Two generalized coordinates used for the planar translation."""
+        return self.coordinates[1:, 0]
+
+    @property
+    def rotation_speed(self):
+        """Generalized speed corresponding to the angular velocity."""
+        return self.speeds[0]
+
+    @property
+    def planar_speeds(self):
+        """Two generalized speeds used for the planar translation velocity."""
+        return self.speeds[1:, 0]
+
+    @property
+    def rotation_axis(self):
+        """The axis about which the rotation occurs."""
+        return self._rotation_axis
+
+    @property
+    def planar_vectors(self):
+        """The vectors that describe the planar translation directions."""
+        return self._planar_vectors
+
+    def _set_vectors(self):
+        # Create list of planar_vectors based on the given planar_vectors
+        planar_vectors = []
+        if isinstance(self._planar_vectors, (tuple, list)):
+            if len(self._planar_vectors) > 2:  # Too many planar vectors
+                raise ValueError(
+                    f'{len(self._planar_vectors)} planar vectors have been '
+                    f'provided, a maximum of 2 is allowed.')
+            for planar_vector in self._planar_vectors:
+                planar_vectors.append(self._axis(planar_vector,
+                                                 self.parent_interframe))
+        elif isinstance(self._planar_vectors, Vector):
+            planar_vectors.append(self._axis(self._planar_vectors,
+                                             self.parent_interframe))
+        elif self._planar_vectors is not None:
+            raise TypeError('planar_vectors should be a tuple or a Vector.')
+
+        # Check if the planar_vectors are not parallel
+        if len(planar_vectors) == 2:
+            n = planar_vectors[0] ^ planar_vectors[1]
```

Comment:
It feels like quite specialist knowledge to know that `^` is overloaded on a `Vector` to compute a cross product. A comment might be useful here.

Reply:
If we're using the `cross` function elsewhere within this method then I think it's clearer to be consistent here too.

Reply:
`cross` is actually a little bit more general, since it uses `^` after checking the type of the first vector. But I agree that `cross` is a lot easier to read.